packages/web/server/queries/complex/assets/index.ts

```diff
@@ -0,0 +1,118 @@
+import { Asset } from "@osmosis-labs/types";
+import { getDisplayDecimalsFromAsset } from "@osmosis-labs/utils";
+import cachified, { CacheEntry } from "cachified";
+import Fuse from "fuse.js";
+import { LRUCache } from "lru-cache";
+
+import { DEFAULT_LRU_OPTIONS } from "~/config/cache";
+import { AssetLists } from "~/config/generated/asset-lists";
+
+import { Search, Sort } from "../parameter-types";
+
+export type GetAssetsParams = {
+  sort?: Partial<Sort>;
+  search?: Search;
+  /** Explicitly match the base or symbol denom. */
+  matchDenom?: string;
+};
+const searchableKeys = ["symbol", "base", "name", "display"];
+
+const cache = new LRUCache<string, CacheEntry>(DEFAULT_LRU_OPTIONS);
+
+/** Get an individual asset explicitly by it's denom (any type) */
+export async function getAsset(
+  anyDenom: string
```

Comment:
What if we convert this too as an object? It will make it easier in the future to add more params if needed.